lib/torrent.js

```diff
@@ -58,6 +58,12 @@ const RECONNECT_WAIT = [1_000, 5_000, 15_000]
 
 const USER_AGENT = `WebTorrent/${VERSION} (https://webtorrent.io)`
 
+// if nodejs or browser that supports FSA
+const FALLBACK_STORE = !process.browser ||
+(typeof navigator !== 'undefined' && navigator.storage?.getDirectory && typeof FileSystemFileHandle !== 'undefined' && FileSystemFileHandle?.prototype?.createWritable)
+  ? FSChunkStore
+  : MemoryChunkStore
```

Comment:
this is a hellscape, how to do this better?

Reply:
This could become:
```js
const FALLBACK_STORE = process.browser &&
  navigator?.storage?.getDirectory &&
  FileSystemFileHandle?.prototype?.createWritable
    ? FSChunkStore
    : MemoryChunkStore;
 ```
But it does assume that in the browser that `navigator` and `FileSystemFileHandle` are both defined, failing that, the undefined checks will still need to exist.

Reply:
this has potential to fail in node...
i considered
```js
const FALLBACK_STORE = !process.browser
  ? FSChunkStore
  : (typeof navigator !== 'undefined' && navigator.storage?.getDirectory && typeof FileSystemFileHandle !== 'undefined' && FileSystemFileHandle?.prototype?.createWritable)
    ? FSChunkStore // FSAChunkStore
    : MemoryChunkStore

Reply:
or a separate 
```js
const SUPPORTS_FSA = (typeof navigator !== 'undefined' && navigator.storage?.getDirectory && typeof FileSystemFileHandle !== 'undefined' && FileSystemFileHandle?.prototype?.createWritable)

const FALLBACK_STORE = !process.browser
  ? FSChunkStore // FSChunkStore - nodejs/bun
  : SUPPORTS_FSA
    ? FSChunkStore // FSAChunkStore - browser
    : MemoryChunkStore
```

Reply:
imo separate would be better at that point